File: lib/models/news_article.dart
class NewsArticle {
  final String id;
  final String title;
  final String source;
  final String? imageUrl;
  final String destinationUrl;
  final String category;
  final DateTime publishedAt;

  NewsArticle({
    required this.id,
    required this.title,
    required this.source,
    this.imageUrl,
    required this.destinationUrl,
    required this.category,
    required this.publishedAt,
  });

  factory NewsArticle.fromJson(Map<String, dynamic> json) {
    return NewsArticle(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      source: json['source'] ?? '',
      imageUrl: json['image_url'],
      destinationUrl: json['destination_url'] ?? '',
      category: json['category'] ?? 'General',
      publishedAt: DateTime.parse(json['published_at'] ?? DateTime.now().toIso8601String()),
    );
  }
}

File: vps_code/src/models/NewsArticle.js
const mongoose = require('mongoose');

const NewsArticleSchema = new mongoose.Schema({
    title: { type: String, required: true },
    source: { type: String, required: true },
    image_url: { type: String },
    destination_url: { type: String, required: true },
    category: { type: String, default: 'General' },
    published_at: { type: Date, default: Date.now },
    is_active: { type: Boolean, default: true },
    sort_order: { type: Number, default: 0 },
    created_at: { type: Date, default: Date.now },
    updated_at: { type: Date, default: Date.now }
});

NewsArticleSchema.pre('save', function(next) {
    this.updated_at = Date.now();
    next();
});

module.exports = mongoose.model('NewsArticle', NewsArticleSchema);
